service/src/main/java/bio/terra/workspace/service/iam/SamService.java

```diff
@@ -300,8 +300,14 @@ public Map<UUID, WsmIamRole> listWorkspaceIdsAndHighestRoles(AuthenticatedUserRe
                   .filter(Objects::nonNull)
                   .collect(Collectors.toList());
           Optional<WsmIamRole> highestRole = WsmIamRole.getHighestRole(workspaceId, roles);
-          // Skip workspaces with no roles. (That means there's a role this WSM doesn't know about.)
+          // Skip workspaces with no roles. (That means there's a role this WSM doesn't know
+          // about.)
           if (highestRole.isPresent()) {
+            // TODO(PF-1875): Support requesting discoverer workspaces in ListWorkspaces.
```

Comment:
Why not just filter in ListWorkspaces, so you don't have to remember to come back to this code?
You need the filtering there anyway.